src/libraries/System.Data.Common/src/System/Data/Common/DbProviderFactories.cs

```diff
@@ -133,6 +137,8 @@ public static bool UnregisterFactory(string providerInvariantName)
             return !string.IsNullOrWhiteSpace(providerInvariantName) && _registeredFactories.TryRemove(providerInvariantName, out _);
         }
 
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026:UnrecognizedReflectionPattern",
+            Justification = "The warning is being display when registering a factory by name or type.")]
```

Comment:
Why is this being suppressed? It seems like this should bubble up to `GetFactory` public methods as well, even though we warn when registering, it probably is best if we warn when getting too.

Reply:
The reason I prefer not to add RUC on all GetFactory is because it's not always unsafe - it's only unsafe when you use assembly qualified name or type when adding. The only truly unsafe is `GetFactory(DataRow)` which always constructs type from the unknown string

Reply:
This makes sense to me. Thanks.